zephyr/include/sof/lib/regions_mm.h

```diff
@@ -10,31 +10,240 @@
 
 #include <adsp_memory.h>
 #include <adsp_memory_regions.h>
+#include <ipc/topology.h>
+#include <rtos/alloc.h>
+#include <sof/list.h>
 #include <zephyr/drivers/mm/system_mm.h>
-#include <zephyr/sys/mem_blocks.h>
 #include <zephyr/init.h>
-#include <ipc/topology.h>
+#include <zephyr/sys/mem_blocks.h>
 
-/*
- * Struct containing information on virtual memory heap.
- * Information about allocated physical blocks is stored in
- * physical_blocks_allocators variable and uses zephyr memblocks api.
+/* Dependency on ipc/topology.h created due to memory capability definitions
+ * that are defined there
+ */
+
+/* Defines maximum amount of memory block allocators in one heap
+ * since minimum realistic block should be DCASHE_LINE size
+ * and block sizes in allocator must be powers of 2
+ * so logicly it gives limited block sizes
+ * eg allocator of block size 64 128 256 512 2048 4096 1024 8192
+ * would end up in 8 allocators.
+ * It is expected that allocations bigger than that would
+ * either be spaned on specifically configured heap
+ */
+#define MAX_MEMORY_ALLOCATORS_COUNT 8
+
+/* list of virtual_memory_heap objects created */
+struct list_item vm_heaps;
+/* list of vmh_active_physical_mapping objects created */
+struct list_item physical_mappings;
+
```

Comment:
I would suggest on prefixing all the variable with vmh.

Better naming:

``
struct list_item vmh_list;
struct list_item vmh_active_phys_list;
``